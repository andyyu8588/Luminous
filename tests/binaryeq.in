class Test {
    public x;
    constructor(x) {
        this.x = x;
    }

    increment() {
        this.x += 1;
        print(this.x);
    }

    decrement() {
        this.x -= 1;
        print(this.x);
    }

    double() {
        this.x *= 2;
        print(this.x);
    }

    negate() {
        this.x /= -1;
        print(this.x);
    }
}

test = Test(1);
test.increment();
test.increment();
test.double();
test.decrement();
test.negate();

function localTest() {
    str = "hello";
    str += " world!";
    print(str);
}

localTest();


innerFunc = null;
function upvalue() {
    val = 3;
    function inner() {
        val += 4;
        val -= 9;
        val /= 2;
        print(val);
    }
    innerFunc = inner;
}

upvalue();
innerFunc();

globalStr = "This is: ";
globalStr += 6;
print(globalStr);

a=[1, "Hello world"];
a[0] += 1;
print(a[0]);
a[1] += "!";
print(a[1]);
