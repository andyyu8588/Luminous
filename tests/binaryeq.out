2
3
6
5
-5
hello world!
-1
This is: 6
